`test_joblib_pickle_across_python_versions_with_mmap` fails on big endian architecture s390x

During joblib packaging for debian, `test_joblib_pickle_across_python_versions_with_mmap` fails on big endian architecture s390x.
This is maybe related to #279, note that  `test_joblib_pickle_across_python_versions` passes.
The full log is available [here](https://ci.debian.net/data/autopkgtest/testing/s390x/j/joblib/26916492/log.gz)

The failure is
```
>                       assert result.dtype == expected.dtype
E                       AssertionError: assert dtype('<i8') == dtype('int64')
E                        +  where dtype('<i8') = memmap([0, 1, 2, 3, 4]).dtype
E                        +  and   dtype('int64') = array([0, 1, 2, 3, 4]).dtype
```

--- a/CHANGES.rst
+++ b/CHANGES.rst
@@ -4,7 +4,8 @@ Latest changes
 In development
 --------------
 
-- TODO
+- Ensure native byte order for memmap arrays when pickling.
+  https://github.com/joblib/joblib/issues/1353
 
 Release 1.2.0
 -------------

--- a/CHANGES.rst
+++ b/CHANGES.rst
@@ -4,7 +4,7 @@ Latest changes
 In development
 --------------
 
-- Ensure native byte order for memmap arrays when pickling.
+- Ensure native byte order for memmap arrays in `joblib.load`.
   https://github.com/joblib/joblib/issues/1353
 
 Release 1.2.0

--- a/joblib/numpy_pickle.py
+++ b/joblib/numpy_pickle.py
@@ -229,7 +229,7 @@ class NumpyArrayWrapper(object):
             )
             warnings.warn(message)
 
-        return marray
+        return _ensure_native_byte_order(marray)
 
     def read(self, unpickler):
         """Read the array corresponding to this wrapper.

